### PR TITLE
Test 3PE lookup APIs

### PR DESCRIPTION
**Pull request by leonerd**
Tests the new API defined in
  https://docs.google.com/document/d/13NGa46a_WWno-XYfe8mQrglQdtOYMFVZtxkPKXDC3ac/edit#

and implemented in synapse in:
  https://github.com/matrix-org/synapse/pull/1026

### CODE REVIEW

File: tests/60app-services/05lookup3pe.pl [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                  field1 => "ONE",
+                  field2 => "TWO",
+               },
+            'fields in received AS request' );
+
+            $request->respond_json( [
+               {
+                  protocol => "protocol",
+                  fields => { field1 => "result" },
+                  userid   => '@remote-user:bridged.example.com',
```
Comment (NegativeMjark) [inline tests/60app-services/05lookup3pe.pl]: The arrows don't line up. :(

Comment (leonerd) [inline tests/60app-services/05lookup3pe.pl]: Done 0aee5ec

File: tests/60app-services/05lookup3pe.pl [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+               }
+            ] );
+
+            Future->done(1);
+         }),
+
+         do_request_json_for( $user,
+            method => "GET",
+            uri    => "/unstable/3pu/protocol",
```
Comment (NegativeMjark) [inline tests/60app-services/05lookup3pe.pl]: If the protocol is an arbitrary protocol for testing can it be something less sensible?

**Comment by leonerd** [comment]
matrixbot retest this please

**Comment by NegativeMjark** [comment]
LGTM

**Comment by leonerd** [comment]
synapse PR is now merged.

**Pull request merged**